tests/loader_regression_tests.cpp

```diff
@@ -3569,3 +3569,38 @@ TEST(DuplicateRegistryEntries, Drivers) {
         "\" from registry \"HKEY_LOCAL_MACHINE\\SOFTWARE\\Khronos\\Vulkan\\Drivers\" to the list due to duplication"));
 }
 #endif
+
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+// Check that valid symlinks do not cause the loader to crash
+TEST(ManifestDiscovery, ValidSymlink) {
+    FrameworkEnvironment env{};
+    env.add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_2_EXPORT_ICD_GPDPA).set_discovery_type(ManifestDiscoveryType::none));
+    env.get_test_icd().physical_devices.push_back({});
+    auto driver_path = env.get_icd_manifest_path(0);
+    fs::path symlink_path = env.get_folder(ManifestLocation::driver_env_var).location() / "symlink_to_driver";
+    env.get_folder(ManifestLocation::driver_env_var).add_existing_file("symlink_to_driver");
+    int res = symlink(driver_path.c_str(), symlink_path.c_str());
+    ASSERT_EQ(res, 0);
+    set_env_var("XDG_CONFIG_DIRS", symlink_path.str());
```

Comment:
Same `XDG_CONFIG_DIRS` question, but apparently this test proves that it works?

Reply:
Hmm that actually means there should be two tests, one with a valid symlink in the folders searched, and one where XDG_CONFIG_DIRS has a valid symlink in the path directly.